deps/rabbit_common/src/rabbit_data_coercion.erl

```diff
@@ -32,7 +32,7 @@ to_atom(Val) when is_binary(Val) -> binary_to_atom(Val, utf8).
 -spec to_atom(Val :: atom() | list() | binary(), Encoding :: atom()) -> atom().
 to_atom(Val, _Encoding) when is_atom(Val)   -> Val;
 to_atom(Val, _Encoding) when is_list(Val)   -> list_to_atom(Val);
-to_atom(Val, Encoding)  when is_binary(Val) -> binary_to_atom(Val, Encoding).
+to_atom(Val, Encoding)  when is_binary(Val) -> binary_to_existing_atom(Val, Encoding).
```

Comment:
This function is used all over the place, we cannot make it throw a `badarg` for unknown atoms.

Reply:
The change doesn't make sense since it leaves another `binary_to_atom` and also the two `list_to_atom` anyway.

If necessary we could have a `to_existing_atom` to be explicit when we know the atom is already defined.

Reply:
ok, so too brave, I'll remove this commit.